staging/src/k8s.io/client-go/rest/request.go

```diff
@@ -365,10 +365,10 @@ func (r *Request) SpecificallyVersionedParams(obj runtime.Object, codec runtime.
 		r.err = err
 		return r
 	}
+	if r.params == nil {
+		r.params = make(url.Values)
+	}
 	for k, v := range params {
-		if r.params == nil {
```

Comment:
there is a change in behavior. previously, if params is empty, r.params will be left untouched, i.e. nil.